cryodrgn/starfile.py

```diff
@@ -128,34 +126,7 @@ def write(self, outstar: str):
             self._write_block(f, self.headers, self.df, block_header="data_")
 
     def get_particles(self, datadir: Optional[str] = None, lazy: bool = True):
-        """
-        Return particles of the starfile
-
-        Input:
-            datadir (str): Overwrite base directories of particle .mrcs
-                Tries both substituting the base path and prepending to the path
-            If lazy=True, returns list of LazyImage instances, else np.array
-        """
-        particles = self.df["_rlnImageName"]
-
-        # format is index@path_to_mrc
-        particles = [x.split("@") for x in particles]
-        ind = [int(x[0]) - 1 for x in particles]  # convert to 0-based indexing
-        mrcs = [x[1] for x in particles]
-        if datadir is not None:
-            mrcs = prefix_paths(mrcs, datadir)
-        for path in set(mrcs):
-            assert os.path.exists(path), f"{path} not found"
-        header = mrc.parse_header(mrcs[0])
-        D = header.D  # image size along one dimension in pixels
-        dtype = header.dtype
-        stride = dtype().itemsize * D * D
-        dataset = [
-            LazyImage(f, (D, D), dtype, 1024 + ii * stride) for ii, f in zip(ind, mrcs)
-        ]
-        if not lazy:
-            dataset = np.array([x.get() for x in dataset])
-        return dataset
+        raise NotImplementedError
```

Comment:
```suggestion
        raise NotImplementedError, "get_particles is deprecated, use the `ImageSource` library."
```

Reply:
I think this syntax for Exceptions is gone in Python 3.7. I'll add a:
```
raise NotImplementedError("get_particles is no longer supported. Use the `ImageSource` library.")
```
 in the next commit.

Reply:
Added in commit 7f3b87558d77091d39877540c5a6906de0ea41cc